src/agent_sdk/inc/du_agent_sdk/agent_module_interface.h

```diff
@@ -12,65 +12,59 @@
 #include "aduc/result.h"
 #include "du_agent_sdk/agent_common.h"
 
-#ifdef __cplusplus
-extern "C"
-{
-#endif /* __cplusplus */
-
-    typedef void* ADUC_AGENT_MODULE_HANDLE;
+EXTERN_C_BEGIN
 
-    typedef ADUC_AGENT_MODULE_HANDLE (*ADUC_AGENT_MODULE_CREATE)(void);
-    typedef void (*ADUC_AGENT_MODULE_DESTROY)(ADUC_AGENT_MODULE_HANDLE);
+typedef void* ADUC_AGENT_MODULE_HANDLE;
+typedef ADUC_AGENT_MODULE_HANDLE (*ADUC_AGENT_MODULE_CREATE)(void);
+typedef void (*ADUC_AGENT_MODULE_DESTROY)(ADUC_AGENT_MODULE_HANDLE);
 
-    typedef struct ADUC_AGENT_CONTRACT_INFO_TAG
-    {
-        const char* provider;
-        const char* name;
-        const int version;
-        const char* contractInfo;
-    } ADUC_AGENT_CONTRACT_INFO;
+typedef struct ADUC_AGENT_CONTRACT_INFO_TAG
+{
+    const char* provider;
+    const char* name;
+    const int version;
+    const char* contractInfo;
+} ADUC_AGENT_CONTRACT_INFO;
 
-    /**
+/**
  * @brief The module data types
  *
  */
-    typedef enum ADUC_MODULE_DATA_TYPE_TAG
-    {
-        ADUC_ModuleDataType_None = 0, /**< No module data*/
-        ADUC_ModuleDataType_String = 1, /**< String data */
-        ADUC_ModuleDataType_Json = 2, /**< JSON string data */
-        ADUC_ModuleDataType_Int64 = 3, /**< 64-bit integer data */
-        ADUC_ModuleDataType_Boolean = 4, /**< Boolean data */
-        ADUC_ModuleDataType_Binary = 5, /**< Binary data */
-    } ADUC_MODULE_DATA_TYPE;
+typedef enum ADUC_MODULE_DATA_TYPE_TAG
```

Comment:
Is this supposed to be the semantic type to cast and interpret as?  e.g. ADUC_ModuleDataType_Json is casting void* as char* and then interpreting it by parsing it as json?  Why is module data needed and what problem does it solve?

Reply:
Please remove if no longer needed.  The moduleData I saw elsewhere is always casting void* to ADU_MQTT_COMMUNICATION_MGR_STATE from what I can tell.